veps/sig-storage/utility-volumes.md

```diff
@@ -0,0 +1,256 @@
+
+3.  **Status Reporting:**
+
+Utility volumes will be reported in the existing VMI `status.volumeStatus` field alongside regular volumes, using the current `VolumeStatus` struct without modifications.
+      * **Target Field Handling:** For utility volumes, the `Target` field will be empty since these volumes are not presented to the VM guest.
+      * **Standard Status Fields:** Utility volumes will use the existing status fields (`Phase`, `Reason`, `Message`, etc.) to report their lifecycle state.
+
+
+### Migration Compatibility
+
+**Design Principle:** Utility volumes are temporary operational tools attached via pod-level attachment pods and are incompatible with libvirt's migration mechanisms. **Consequently, these volumes will not migrate with the VM.**
+
+**Migration Process:**
+1. **Pre-Migration:** In the unlikely event that a utility operation is active at migration time, each utility type must be designed to handle this possibility, either aborting the operation or ensuring its completion before migration proceeds.
+2. **During Migration:** Utility volumes remain attached to source VM. Migration controller filters utility volumes preventing the creation of attachment pods on destination node, namespace or cluster.
+3. **Migration Completion:** Just before migration transitions to succeeded state, utility volumes are automatically removed from both VMI and VM specifications to prevent post-migration re-attachment.
+4. **Post-Migration:** No utility volumes exist on destination node. Manual reattachment is possible if needed and storage and access mode allows (e.g., RWO PVCs may need time to detach from source node).
+
+**Rationale:** This approach avoids complex migration logic while ensuring no functional loss, since utility volumes cannot operate during migration anyway (e.g., memory dumps impossible while VM memory is transferring, same for VM backups).
+
+## API Examples
+
+### Add a utility volume
+
+#### Before:
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: my-vm
+spec:
+  running: true
+  template:
+    spec:
+      domain:
+        # VM domain configuration
+      volumes:
+        # Existing volumes for the guest VM
+        - name: system-disk
+          containerDisk:
+            image: kubevirt/fedora-cloud-container-disk-demo:latest
+      # No utility volumes initially
+```
+
+#### After (adding a memoryDump utility volume):
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: my-vm
+spec:
+  running: true
+  template:
+    spec:
+      domain:
+        # remains unchanged
+      volumes:
+        # remains unchanged
+        - name: system-disk
+          containerDisk:
+            image: kubevirt/fedora-cloud-container-disk-demo:latest
+      # NEW FIELD:
+      utilityVolumes:
+        - name: memorydump-target
```

Comment:
one edge case to consider, given the current hotplug implementation you have to make sure to not have name collisions with `spec.volumes` and `spec.utilityVolumes` or work around that somehow internally